utilities/transactions/transaction_lock_mgr.cc

```diff
@@ -377,12 +377,16 @@ void TransactionLockMgr::DecrementWaitersImpl(
 }
 
 bool TransactionLockMgr::IncrementWaiters(
-    const TransactionImpl* txn, const autovector<TransactionID>& wait_ids) {
+    const TransactionImpl* txn, const autovector<TransactionID>& wait_ids,
+    const std::string& key, const uint32_t& cf_id) {
   auto id = txn->GetID();
-  std::vector<TransactionID> queue(txn->GetDeadlockDetectDepth());
+  std::vector<std::pair<TransactionID, int>> queue(
```

Comment:
Instead of using std::pair, maybe it'd be easier to read if we just made a second vector of the same size storing backtracking info?